app/models/game.rb

```diff
@@ -28,6 +28,27 @@ def board_state
     board
   end
 
+  def opp_color(color)
+    color == 'white' ? 'black' : 'white'
+  end
+
+  def in_check?(color)
+    # checking king
+    king = pieces.find_by(type: 'King', color: color)
+    # checking pieces
+    opponent = pieces.where(captured: false) # self.pieces.where(color: !color, captured: false)
```

Comment:
here you could add `where(captured: false, color: opp_color(color)`

Then you won't need the `opp_color` calls later